### PR TITLE
Updated http source documentation

### PR DESCRIPTION
**Pull request by asifsmohammed**
Signed-off-by: Asif Sohail Mohammed <nsifmoh@amazon.com>

### Description
- Updated HTTP source ssl documentation
 
### Check List
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]
Thanks for taking note of this!

File: data-prepper-plugins/http-source/README.md [hunk 1]
```diff
@@ -60,9 +60,14 @@ create a plugin which implements [`ArmeriaHttpAuthenticationProvider`](../armeri
 
 ### SSL
 
-* ssl(Optional) => A `boolean` enables TLS/SSL. Default is ```false```.
-* ssl_certificate_file(Optional) => A `String` represents the SSL certificate chain file path. Required if ```ssl``` is set to ```true```.
-* ssl_key_file(Optional) => A `String` represents the SSL key file path. Only decrypted key file is supported. Required if ```ssl``` is set to ```true```.
+* ssl(Optional) => A `boolean` that enables TLS/SSL. Default is ```false```.
+* ssl_certificate_file(Optional) => A `String` that represents the SSL certificate chain file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is set to `true`.
```
Comment (oeyh) [inline data-prepper-plugins/http-source/README.md:64]: Relaying a message from another PR: Is `ssl_certificate_file` required only when `ssl` is true _and_ `use_acm_certificate_for_ssl` is false?

**Review by oeyh — COMMENTED** [review]
Thanks for adding these! I will update the documentation website accordingly as well.

File: data-prepper-plugins/http-source/README.md (hunk 1 above)
Comment (asifsmohammed) [reply data-prepper-plugins/http-source/README.md:64]: With `ssl` enabled by default, they are no longer required as `ssl_certificate_file` and `ssl_key_file` have default values.

Comment (dlvenable) [reply data-prepper-plugins/http-source/README.md:64]: Yes, we should clarify that here in the documentation.

**Review by oeyh — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**